src/git.jl

```diff
@@ -1,32 +1,56 @@
 function _git_cmd(str::AbstractString;
                   adjust_PATH::Bool = true,
                   adjust_LIBPATH::Bool = true)
-    git_path, env_mapping = _env_mapping(; adjust_PATH, adjust_LIBPATH)
-    return addenv(`$(git_path) $(split(str))`, env_mapping)
+    cmd = git(; adjust_PATH, adjust_LIBPATH)
+    return `$(cmd) $(split(str))`
 end
 
 macro git_cmd(ex)
     return _git_cmd(ex)
 end
 
-function _gitrepl_parser(repl_input::AbstractString)
-    return quote
-        GitCommand.git() do git
-            repl_input = $(Expr(:quote, repl_input))
-            run(`$(git) $(split(repl_input))`)
-            return nothing
-        end
+function git(; adjust_PATH::Bool = true, adjust_LIBPATH::Bool = true)
+    if Sys.iswindows()
+        return Git_jll.git(; adjust_PATH, adjust_LIBPATH)
     end
-end
 
-function git(;
-             adjust_PATH::Bool = true,
-             adjust_LIBPATH::Bool = true)
-    git_path, env_mapping = _env_mapping(; adjust_PATH, adjust_LIBPATH)
-    return addenv(`$(git_path)`, env_mapping...)
+    root = Git_jll.artifact_dir
+
+    libexec = joinpath(root, "libexec")
+    libexec_git_core = joinpath(libexec, "git-core")
+
+    share = joinpath(root, "share")
+    share_git_core = joinpath(share, "git-core")
+    share_git_core_templates = joinpath(share_git_core, "templates")
+
+    libpaths = [
+        dirname(Expat_jll.libexpat_path),
+        dirname(Gettext_jll.libgettext_path),
+        dirname(LibCURL_jll.libcurl_path),
+        dirname(Libiconv_jll.libiconv_path),
+        dirname(XML2_jll.libxml2_path),
+        get(ENV, JLLWrappers.LIBPATH_env, "")
+    ]
+    filter!(x -> !isempty(x), libpaths)
+    newlibpath = join(libpaths, _separator())
+
+    ssl_cert = joinpath(dirname(Sys.BINDIR), "share", "julia", "cert.pem")
+
+    env_mapping = Dict{String,String}()
+    env_mapping["GIT_EXEC_PATH"] = libexec_git_core
+    env_mapping["GIT_SSL_CAINFO"] = ssl_cert
+    env_mapping["GIT_TEMPLATE_DIR"] = share_git_core_templates
+    env_mapping[JLLWrappers.LIBPATH_env] = newlibpath
+    if haskey(ENV, "PATH")
+        env_mapping["PATH"] = ENV["PATH"]
+    end
```

Comment:
`addenv` by default inherits the current environment, so also this is redundant

Reply:
Done - I've removed those lines.